polaris/dataset/converters/_sdf.py

```diff
@@ -103,25 +103,31 @@ def _get_name(mol: dm.Mol):
         if self.mol_prop_as_cols:
             props &= ~Chem.PropertyPickleOptions.MolProps
         bytes_data = [mol.ToBinary(props) for mol in df[tmp_col]]
-
         df.drop(columns=[tmp_col], inplace=True)
 
-        # Create the zarr array
-        factory.zarr_root.array(self.mol_column, bytes_data, dtype=bytes, chunks=self.chunks)
+        if self.mol_column in factory.zarr_root and append:
+            # append existing array
+            pointer_start = factory.zarr_root[self.mol_column].shape[0]
+            factory.zarr_root[self.mol_column].append(bytes_data)
+        else:
+            # Create the zarr array
+            factory.zarr_root.array(self.mol_column, bytes_data, dtype=bytes, chunks=self.chunks)
+            pointer_start = 0
 
         # Add a pointer column to the table
         # We support grouping by a key, to allow inputs of variable length
         grouped = pd.DataFrame(columns=[*df.columns, self.mol_column])
         if self.groupby_key is not None:
             for _, group in df.reset_index(drop=True).groupby(by=self.groupby_key):
-                start = group.index[0]
-                end = group.index[-1]
-
-                if group.nunique().sum() != len(group.columns):
-                    raise ValueError(
-                        f"After grouping by {self.groupby_key}, values for other columns are not unique within a group. "
-                        "Please handle this manually to ensure aggregation is done correctly."
-                    )
+                start = pointer_start + group.index[0]
+                end = pointer_start + group.index[-1]
+                # Lu: is the check neccessary ?
+                # if group.nunique().sum() != len(group.columns): # deduct group column
+                #     print(group)
+                #     raise ValueError(
+                #         f"After grouping by {self.groupby_key}, values for other columns are not unique within a group. "
+                #         "Please handle this manually to ensure aggregation is done correctly."
+                #     )
```

Comment:
**Comment:** As discussed in person - This check is indeed necessary because in these cases you need to come up with a way to manually group the data. However, people are not required to use the factory, so this is not a blocker.